homework-4/Program.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CollectionApplication
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Welcome to the Collection Application!");
+
+            while (true)
+            {
+                Console.WriteLine("\nPlease enter a number from the given menu:");
+                Console.WriteLine("1. Continue the application using List.");
+                Console.WriteLine("2. Continue the application using Dictionary.");
+                Console.WriteLine("3. Exit the application.");
+
+                if (!int.TryParse(Console.ReadLine(), out int choice))
+                {
+                    Console.WriteLine("Please enter a valid number.");
+                    continue;
+                }
+
+                switch (choice)
+                {
+                    case 1:
+                        ListMenu();
+                        break;
+                    case 2:
+                        DictionaryMenu();
+                        break;
+                    case 3:
+                        Console.WriteLine("Exiting the application...");
+                        return;
+                    default:
+                        Console.WriteLine("Please enter a valid number.");
+                        break;
+                }
+            }
+        }
+
+        static void ListMenu()
+        {
+            var collection = new CollectionList();
+            collection.Menu();
+        }
+
+        static void DictionaryMenu()
+        {
+            var collection = new CollectionDictionary();
+            collection.Menu();
+        }

```

Comment:
You created these two functions just to create the object and for calling the menu.. what you could have done is.. create a interface `ICollectionOperation` which will have the definitions of all the operation that CollectionDictionary and CollectionList has.. and create the loosely coupled object of CollectionList or CollectionDictionary itself in the main function and call the menu function in main directly.

For eg.. 

```
static void Main(string[] args) {
    // rest of the code
    
    while (true) {
       // rest of the code
       
       switch (input)
                {
                    case "1":
                        ICollectionOperation collectionOperation = new ListOperation();
                        collectionOperation.showMenu();
                        break;
                    case "2":
                        ICollectionOperation collectionOperation = new DictionaryOperation();
                        collectionOperation.showMenu();
                        break;
                    case "3":
                        return;
                    default:
                        Console.WriteLine("Please enter a valid number.");
                        break;
                }
   }
}
```